Patches/Rimworld-Style Pilas And Bows Strapped With Grenades and Shells Extended/Arrows.xml

```diff
@@ -0,0 +1,803 @@
+                        <label>arrow (molotov)</label>
+                        <graphicData>
+                            <texPath>Bowlotov</texPath>
+                            <graphicClass>Graphic_Single</graphicClass>
+                        </graphicData>
+                        <ammoClass>ArrowMolotov</ammoClass>
+                        <statBases>
+                            <MarketValue>7.18</MarketValue>
+                        </statBases>
+                        <tradeTags>
+                            <li>CE_AutoEnableCrafting_CraftingSpot</li>
+                        </tradeTags>
+                    </ThingDef>
+                    
+                    <ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoArrowBase">
+                        <defName>Ammo_Arrow_EMP</defName>
+                        <label>arrow (EMP)</label>
+                        <graphicData>
+                            <texPath>BowMP</texPath>
+                            <graphicClass>Graphic_Single</graphicClass>
+                        </graphicData>
+                        <ammoClass>ArrowEMP</ammoClass>
+                        <statBases>
+                            <MarketValue>21.22</MarketValue>
+                        </statBases>
+                        <tradeTags>
+                            <li>CE_AutoEnableCrafting_CraftingSpot</li>
+                        </tradeTags>
+                    </ThingDef>
+                    
+                    <ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoArrowBase">
+                        <defName>Ammo_Arrow_Battery</defName>
+                        <label>arrow (battery)</label>
+                        <graphicData>
+                            <texPath>Bowttery</texPath>
+                            <graphicClass>Graphic_Single</graphicClass>
+                        </graphicData>
+                        <ammoClass>ArrowBattery</ammoClass>
+                        <statBases>
+                            <MarketValue>26.04</MarketValue>
+                        </statBases>
+                        <tradeTags>
+                            <li>CE_AutoEnableCrafting_CraftingSpot</li>
+                        </tradeTags>
+                    </ThingDef>
+                </value>
+            </li>
+
+            <!--Great Arrows-->
+            <li Class="PatchOperationAdd">
+                <xpath>/Defs</xpath>
+                <value>
+                    <ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoGreatArrowBase">
+                        <defName>Ammo_GreatArrow_Grenade</defName>
+                        <label>great arrow (grenade)</label>
+                        <graphicData>
+                            <texPath>Bownade</texPath>
+                            <graphicClass>Graphic_Single</graphicClass>
+                        </graphicData>
+                        <ammoClass>ArrowGrenade</ammoClass>

```

Comment:
Would be good to add `<generateAllowChance>0</generateAllowChance>` to some of these ammo types with an appropriate value. This is what controls how frequently they generate in NPC pawn loadouts and runs from 1 (standard rate) to 0 (don't appear).

Reply:
Would `</generateCommonality>` also work for the pilas or should I stick to the `<generateAllowChance>`?

Reply:
Hrmm, that's a good question--since the pila is both a weapon *and* ammo. I think `generateAllowChance` may be the way to go here, but am not entirely certain.